scripts/safe_template_linter.py

```diff
@@ -863,69 +766,247 @@ def _parse_string(self, template, start_index):
                 }
 
 
-class UnderscoreTemplateLinter(object):
-    """
-    The linter for Underscore.js template files.
+class BaseLinter(object):
     """
+    BaseLinter provides some helper functions that are used by multiple linters.
 
-    _skip_underscore_dirs = _skip_dirs + ('test',)
+    """
+    def __init__(self):
+        """
+        Init method.
+        """
+        self._skip_dirs = (
+            '.pycharm_helpers',
+            'common/static/xmodule/modules',
+            'node_modules',
+            'reports/diff_quality',
+            'spec',
+            'scripts/tests/templates',
+            'test_root',
+            'vendor',
+        )
 
-    def process_file(self, directory, file_name):
+    def _is_skip_dir(self, skip_dirs, directory):
         """
-        Process file to determine if it is an Underscore template file and
-        if it is safe.
+        Determines whether a directory should be skipped or linted.
 
         Arguments:
-            directory (string): The directory of the file to be checked
-            file_name (string): A filename for a potential underscore file
+            skip_dirs: The configured directories to be skipped.
+            directory: The current directory to be tested.
 
         Returns:
-            The file results containing any violations.
+             True if the directory should be skipped, and False otherwise.
 
         """
-        full_path = os.path.normpath(directory + '/' + file_name)
-        results = FileResults(full_path)
-
-        if not self._is_valid_directory(directory):
-            return results
-
-        if not file_name.lower().endswith('.underscore'):
-            return results
-
-        return self._load_and_check_underscore_file_is_safe(full_path, results)
+        for skip_dir in skip_dirs:
+            dir_contains_skip_dir = '/' + skip_dir + '/' in directory
+            if dir_contains_skip_dir or directory.startswith(skip_dir) or directory.endswith(skip_dir):
```

Comment:
Seems like a regex would work better here.  This doesn't ensure that the directory ends with "/"+skip_dir, which it should?